cd/auto-update/auto-update-controllers.sh

```diff
@@ -0,0 +1,191 @@
+PR_TARGET_BRANCH=${PR_TARGET_BRANCH:-$DEFAULT_PR_TARGET_BRANCH}
+
+LOCAL_GIT_BRANCH="main"
+
+DEFAULT_GITHUB_ISSUE_ORG="aws-controllers-k8s"
+GITHUB_ORG=${GITHUB_ORG:-$DEFAULT_GITHUB_ISSUE_ORG}
+
+DEFAULT_GITHUB_ISSUE_REPO="community"
+GITHUB_ISSUE_REPO=${GITHUB_ISSUE_REPO:-$DEFAULT_GITHUB_ISSUE_REPO}
+
+GITHUB_ISSUE_ORG_REPO="$GITHUB_ORG/$GITHUB_ISSUE_REPO"
+
+DEFAULT_GITHUB_LABEL="ack-bot-autogen"
+GITHUB_LABEL=${GITHUB_LABEL:-$DEFAULT_GITHUB_LABEL}
+
+DEFAULT_GITHUB_LABEL_COLOR="3C6110"
+GITHUB_LABEL_COLOR=${GITHUB_LABEL_COLOR:-$DEFAULT_GITHUB_LABEL_COLOR}
+
+# Check all the dependencies are present in container.
+source "$TEST_INFRA_DIR"/scripts/lib/common.sh
+source "$CD_DIR"/lib/gh.sh
+check_is_installed git
+check_is_installed gh
+
+USER_EMAIL="${GITHUB_ACTOR}@users.noreply.${GITHUB_DOMAIN:-"github.com"}"
+if [ -n "${GITHUB_EMAIL_PREFIX}" ]; then
+    USER_EMAIL="${GITHUB_EMAIL_PREFIX}+${USER_EMAIL}"
+fi
+
+# set the GitHub configuration for using GitHub cli.
+git config --global user.name "${GITHUB_ACTOR}" >/dev/null
+git config --global user.email "${USER_EMAIL}" >/dev/null
+
+# Find the controller-bootstrap semver from the latest tag on the controller-bootstrap repo
+ACK_CONTROLLER_BOOTSTRAP_VERSION=$(git describe --tags --always --dirty)
+
+DEFAULT_PR_SOURCE_BRANCH="ack-bot/controller-bootstrap-$ACK_CONTROLLER_BOOTSTRAP_VERSION"
+PR_SOURCE_BRANCH=${PR_SOURCE_BRANCH:-$DEFAULT_PR_SOURCE_BRANCH}
+
+# find all the directories whose name ends with 'controller'
+pushd "$WORKSPACE_DIR" >/dev/null
+  CONTROLLER_NAMES=$(find . -maxdepth 1 -name "*-controller" -type d | cut -d"/" -f2)
+popd >/dev/null
+
+for CONTROLLER_NAME in $CONTROLLER_NAMES; do
+  SERVICE_NAME=$(echo "$CONTROLLER_NAME"| sed 's/-controller$//g')
+  CONTROLLER_DIR="$WORKSPACE_DIR/$CONTROLLER_NAME"
+  cd "$CONTROLLER_BOOTSTRAP_DIR"
+
+  echo -n "auto-update-controllers.sh][INFO] Ensuring that GitHub label $GITHUB_LABEL exists for $GITHUB_ORG/$CONTROLLER_NAME ... "
+  if ! gh api repos/"$GITHUB_ORG"/"$CONTROLLER_NAME"/labels/"$GITHUB_LABEL" --silent >/dev/null; then
+    echo ""
+    echo "auto-update-controllers.sh][INFO] Could not find label $GITHUB_LABEL in repo $GITHUB_ORG/$CONTROLLER_NAME"
+    echo -n "Creating new GitHub label $GITHUB_LABEL ... "
+    if ! gh api -X POST repos/"$GITHUB_ORG"/"$CONTROLLER_NAME"/labels -f name="$GITHUB_LABEL" -f color="$GITHUB_LABEL_COLOR" >/dev/null; then
+      echo ""
+      echo "auto-update-controllers.sh][ERROR] Failed to create label $GITHUB_LABEL. Skipping $CONTROLLER_NAME"
+      continue
+    else
+      echo "ok"
+    fi
+  else
+    echo "ok"
+  fi
+
+  echo "auto-update-controllers.sh][INFO] Updating project description files of existing controller using command 'make run'"
+  export SERVICE=$SERVICE_NAME
+  MAKE_RUN_OUTPUT_FILE=/tmp/"$SERVICE_NAME"_make_run_output
+  MAKE_RUN_ERROR_FILE=/tmp/"$SERVICE_NAME"_make_run_error
+  if ! make run > "$MAKE_RUN_OUTPUT_FILE" 2> "$MAKE_RUN_ERROR_FILE"; then
```

Comment:
Where does `make run` come from in the context of this script's execution?

Reply:
In the context of this script’s execution, the `make run` command is run in each loop iteration from the [controller-bootstrap](https://github.com/aws-controllers-k8s/test-infra/pull/240/files#diff-72d8a26c5de0f03f2064a5baf611d612b8157394cd8583e692f26e14836d7f67R85) make target, `[run](https://github.com/aws-controllers-k8s/controller-bootstrap/blob/main/Makefile#L59-L62)`.  The `make run` command can be modified to an alternative name, or I can add a make target specifically for the `update` command in the controller-bootstrap. Thank you!

Reply:
:+1: